src/qibolab/sweeper.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy.typing as npt
+
+
+@dataclass
+class Sweeper:
```

Comment:
I know I have also not done this in my branch, but we should document this and provide at least one usage example. I left it for the end so that the API is finalized, but we are now almost ready to merge so it is time to do so.